beacon-chain/rpc/validator_server.go

```diff
@@ -150,26 +162,66 @@ func (vs *ValidatorServer) CommitteeAssignment(
 		}
 	}
 
-	committee, shard, slot, isProposer, err :=
-		helpers.CommitteeAssignment(beaconState, req.EpochStart, uint64(idx), false)
-	if err != nil {
-		return nil, err
-	}
-	vsr, err := vs.ValidatorStatus(ctx, &pb.ValidatorIndexRequest{PublicKey: req.PublicKey[0]})
-	if err != nil {
-		return nil, err
+	for slot := req.EpochStart; slot < req.EpochStart+params.BeaconConfig().SlotsPerEpoch; slot++ {
+		if featureconfig.FeatureConfig().EnableCommitteesCache {
+			if exists, committees, err := vs.committeesCache.CommitteesInfoBySlot(int(slot)); exists || err != nil {
+				if err != nil {
+					return nil, err
+				}
+				span.AddAttributes(trace.BoolAttribute("committeeCacheHit", true))
+				committee, shard, slot, isProposer, err :=
+					helpers.ValidatorAssignment(index, slot, committees.committees)
+				if err == nil {
```

Comment:
what of err != nil, the error isnt returned here

Reply:
the only error is `fmt.Errorf("unable to find assignment for validator %d at slot %d", vIndex, slot)` that just means cache miss and we'll add it to the cache below